scipio/src/executor.rs

```diff
@@ -717,6 +818,34 @@ impl<T> Task<T> {
         }
     }
 
+    /// Returns the id of the current executor
+    ///
+    /// If called from a [`LocalExecutor`], returns the id of the executor.
+    ///
+    /// Otherwise, this method panics.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::{LocalExecutor, Task};
+    ///
+    /// let local_ex = LocalExecutor::new(None).expect("failed to create local executor");
+    ///
+    /// local_ex.run(async {
+    ///     println!("my ID: {}", Task::<()>::id());
+    /// });
+    /// ```
+    pub fn id() -> usize
```

Comment:
Having `later()` or `local()` be proxies to the `LocalExecutor` is fine since those two methods either deal with or return a new `Task`. This one is just confusing. 

I think we should have a way to access the local executor (ala `engine()`). Those utility functions, in turn, should call into that.

Reply:
The problem is that in Rust, you can only safely use thread local things inside a scope. You cannot have an `engine()` function like seastar. At most you could have `with_engine(Function)`. (see the uses of `with`). Between those two things I prefer to provide individual functions.